diffmpm/constraint.py

```diff
@@ -15,20 +26,44 @@ def tree_unflatten(cls, aux_data, children):
         del children
         return cls(*aux_data)
 
-    def apply(self, obj, ids):
+    def apply_vel(self, vel, ids):
+        """Apply constraint values to the passed object.
+
+        Parameters
+        ----------
+        obj : diffmpm.node.Nodes, diffmpm.particle._ParticlesState
```

Comment:
I think it should be vel in the docstring or obj in the function, same issue with all the functions.